infra/elasticsearch/src/vectorizer/index.ts

```diff
@@ -1,22 +1,34 @@
-// vectorizer is imported by code-du-travail-api which is using CommonJS, and throwing an exception
-// when requiring code-du-travail-data ES module, thus we keep using CommonJS import here
-const fetch = require("node-fetch");
-const { stopwords: semantic_stopwords } = require("../dataset/stop_words");
+//
 
+import fetch from "node-fetch";
+
+import { stopwords } from "../dataset/stop_words";
 // URL of the TF serve deployment
-const NLP_URL =
-  process.env.NLP_URL ||
-  "https://preprod-serving-ml.dev2.fabrique.social.gouv.fr";
-const tfServeURL = NLP_URL + "/v1/models/sentqam:predict";
 
-function stripAccents(text) {
+function stripAccents(text: string) {
   // strip accents
   return text.normalize("NFD").replace(/[\u0300-\u036f]/g, "");
 }
 
-const stopWords = new Set(semantic_stopwords.map(stripAccents));
+async function callTFServe(body: string): Promise<number[][]> {
+  const NLP_URL =
+    process.env.NLP_URL ??
+    "https://preprod-serving-ml.dev2.fabrique.social.gouv.fr";
+
+  const tfServeURL = NLP_URL + "/v1/models/sentqam:predict";
+
+  const response = await fetch(tfServeURL, { body, method: "POST" });
+  if (response.ok) {
+    const json: { outputs: number[][] } = await response.json();
+    return json.outputs;
+  } else {
+    throw new Error(response.statusText);
+  }
+}
+
+export function preprocess(text: string): string {
+  const stopWords = new Set(stopwords.map(stripAccents));
```

Comment:
this should be outside of the function there is no need to recreate the stopwords list

Reply:
I would argue that less globales is better but ok.
I will restore the line 👍